### PR TITLE
Nexus Mutual - products & covers

### PR DESCRIPTION
**Pull request by tomfutago**
### Contribution type
Please check the type of contribution this pull request is for:

- [x] New spell(s)
- [ ] Adding to existing spell lineage
- [ ] Bug fix

---

### For new spell(s)
If you are building new spell(s), please provide the following information:

- **Spell name(s):**
  - `nexusmutual_ethereum_products_v1`
  - `nexusmutual_ethereum_product_types_v2`
  - `nexusmutual_ethereum_products_v2`
  - `nexusmutual_ethereum_covers_v1`
  - `nexusmutual_ethereum_covers_v2`
- **Description:** base spells retrieving products offered by Nexus Mutual & covers bought to date
- **Who are the new spell(s) for?** internal team and/or general community
- **How will the new spell(s) be used downstream?** improvements to existing dashboards
- **Implementation details:** all new spells are materialised as views (except for one static table) and might need revisiting down the line to convert to incremental loads
- **Test instructions:** no specific instructions here apart from CI taking care of implemented basic tests like columns nullability and uniqueness

---

### CODE REVIEW

File: models/nexusmutual/ethereum/covers/nexusmutual_ethereum_product_information.sql [hunk 1]
```diff
@@ -0,0 +1,16 @@
+{{ config(
+    alias = 'product_information',
```
Comment (jeff-dude) [inline models/nexusmutual/ethereum/covers/nexusmutual_ethereum_product_information.sql:2]: please include `schema` in each model config block
we will eventually remove from project file

File: models/nexusmutual/ethereum/covers/nexusmutual_ethereum_product_information.sql [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+    materialized = 'view',
+    post_hook='{{ expose_spells(\'["ethereum"]\',
```
Comment (jeff-dude) [inline models/nexusmutual/ethereum/covers/nexusmutual_ethereum_product_information.sql:4]: do all spells in this PR need to be exposed on data explorer? or can we narrow down to just a few?

**Review by jeff-dude — CHANGES_REQUESTED** [review]

**Comment by jeff-dude** [comment]
another consideration, but not a must:
if you expect to deploy on other chains, maybe it makes sense to reorg directory to `models/nexusmutual/<use case>/<blockchain>/` so it's easier to stick to one area when navigating directories, rather than seeing multiple use cases with a chain directory. hopefully that makes sense 😅

File: models/nexusmutual/ethereum/covers/nexusmutual_ethereum_product_information.sql (hunk 2 above)
Comment (tomfutago) [reply models/nexusmutual/ethereum/covers/nexusmutual_ethereum_product_information.sql]: updated 2 old ones, all the rest - yes please

**Comment by tomfutago** [comment]
> another consideration, but not a must: if you expect to deploy on other chains, maybe it makes sense to reorg directory to `models/nexusmutual/<use case>/<blockchain>/` so it's easier to stick to one area when navigating directories, rather than seeing multiple use cases with a chain directory. hopefully that makes sense 😅

I like the idea with dir reorg, but there are no plans to deploy on other chains, so if it's only `ethereum` it probably makes more sense to keep all <use case> folders where they are. btw, more of them are coming soon..

**Comment by tomfutago** [comment]
CI is not too happy last few days (on any PR not just this one), keeps blowing up on `"Error listing tables for catalog hive` msg..

**Comment by jeff-dude** [comment]
> CI is not too happy last few days (on any PR not just this one), keeps blowing up on `"Error listing tables for catalog hive` msg..

yeah, i'll have to investigate this one. the `dbt run` sends a bunch of queries to the db to "list" schemas for some reason prior to actually running anything. it appears to be getting stuck here

**Review by jeff-dude — APPROVED** [review]
code looks good, i'll keep trying CI. reruns have helped other PRs. we are working with our platform team on how to fix

**Comment by jeff-dude** [comment]
> code looks good, i'll keep trying CI. reruns have helped other PRs. we are working with our platform team on how to fix

ugh. keeps happening. is this in a rush? can it wait until we find the resolution?

**Comment by tomfutago** [comment]
> > code looks good, i'll keep trying CI. reruns have helped other PRs. we are working with our platform team on how to fix
> 
> ugh. keeps happening. is this in a rush? can it wait until we find the resolution?

not a massive rush, happy to wait

**Comment by jeff-dude** [comment]
finally!

**Pull request merged**